test/services/database/challenge_repository_test.dart

```diff
@@ -0,0 +1,206 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:proxima/models/database/challenge/challenge_data.dart";
+import "package:proxima/models/database/challenge/challenge_firestore.dart";
+import "package:proxima/models/database/post/post_data.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/services/database/challenge_repository_service.dart";
+import "package:proxima/services/database/post_repository_service.dart";
+
+import "../../mocks/data/firestore_user.dart";
+import "../../mocks/data/geopoint.dart";
+import "../../mocks/data/post_data.dart";
+
+Future<List<PostData>> addPosts(
+  PostRepositoryService postRepository,
+  GeoPoint pos,
+  int n,
+) async {
+  final fakePosts = PostDataGenerator.generatePostData(n);
+  for (final post in fakePosts) {
+    await postRepository.addPost(post, pos);
+  }
+  return fakePosts;
+}
+
+void main() {
+  late FakeFirebaseFirestore firestore;
+  late PostRepositoryService postRepository;
+  late ChallengeRepositoryService challengeRepository;
+
+  setUp(() async {
+    firestore = FakeFirebaseFirestore();
+    postRepository = PostRepositoryService(firestore: firestore);
+    challengeRepository = ChallengeRepositoryService(
+      firestore: firestore,
+      postRepositoryService: postRepository,
+    );
+  });
+
+  const userPos = userPosition1;
+  final inChallengeRange = GeoPointGenerator.createOnEdgeInsidePosition(
+    userPos,
+    ChallengeRepositoryService.maxChallengeRadius,
+  );
+
+  final uid = testingUserFirestoreId;
+
+  group("ChallengeRepositoryService", () {
```

Comment:
I'm not fund of this group name. I don't think we used PascalCase for test naming elsewhere. Moreover, putting all the tests of the file in a group seems to diminish the concept of group in my opinion. I think it is worth trying to split it up, though I'm not sure I see a good way to do it, but maybe the following could make sense:
- Challenge repository getter
- Challenge repository challenge update
- Challenge repository finish and delete

Reply:
Done with 1c06dc7e0ba2671444797c58ed83e9a7de88ccc1